src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/Protocol.java

```diff
@@ -65,7 +66,9 @@ public Protocol(Client client, QueryAction queryAction, Object queryResult, Stri
     public Protocol(Exception e) {
         this.formatType = null;
         this.status = ERROR_STATUS;
-        this.error = new ErrorMessage(e, ERROR_STATUS);
+        this.error = e instanceof ElasticsearchException ? new ElasticsearchErrorMessage((ElasticsearchException) e,
+                ((ElasticsearchException) e).status().getStatus())
```

Comment:
Because `ElasticsearchErrorMessage` is dedicated to `ElasticsearchException`, I think you can pass in `e` only and get everything you need inside the constructor.